### PR TITLE
[NCL-4117] restricting of editing Product Version name due to having closed milestone

### PR DESCRIPTION
**Pull request by michalovjan**
### Checklist:

* [NCL-4117] added restriction on a user when trying to update product version name in the scenario of version having a closed milestone (both REST and UI)
* [NCL-4117] added integration test

### CODE REVIEW

**Comment by thescouser89** [comment]
retest this please

**Comment by michalovjan** [comment]
retest this please

File: rest/src/main/java/org/jboss/pnc/rest/endpoint/ProductVersionEndpoint.java [hunk 1]
```diff
@@ -134,7 +134,8 @@ public Response getSpecific(
     @Path("/{id}")
     public Response update(@ApiParam(value = "Product Version id", required = true) @PathParam("id") Integer id,
             ProductVersionRest productVersionRest) throws RestValidationException {
-        return super.update(id, productVersionRest);
+        productVersionProvider.update(id, productVersionRest);
```
Comment (jbartece) [inline rest/src/main/java/org/jboss/pnc/rest/endpoint/ProductVersionEndpoint.java]: You should not update the endpoint, but rather create a validator and bind it to the Rest entity.
See RepositoryConfigurationEndpoint#update RepositoryConfigurationRest.internalUrl ScmUrl ScmUrlValidator

**Comment by michalovjan** [comment]
retest this please

**Comment by michalovjan** [comment]
After implementing the issue through BeanValidation i've found out that this is not high enough level to validate, because we need access to database which is not available on the bean level. -> leaving this on rest level

**Comment by michalovjan** [comment]
retest this please

**Comment by michalovjan** [comment]
retest this please

**Comment by thescouser89** [comment]
retest this please

**Pull request merged**